src/cmd/embeddedrules/rules.php

```diff
@@ -92,6 +92,46 @@ function precedence() {
     $_ !== $_ | $_;
     $_ | $_ !== $_;
   }
+
+  /** @warning === has higher precedence than ?? */
+  any_eq3_coalesce: {
```

Comment:
I think we don't need `any_*` groups of 1 pattern.
Or there is some reason behind it?